agent/agent.go

```diff
@@ -1984,13 +2017,116 @@ func (a *Agent) purgeCheck(checkID types.CheckID) error {
 	return nil
 }
 
+// persistedServiceConfig is used to serialize the resolved service config that
+// feeds into the ServiceManager at registration time so that it may be
+// restored later on.
+type persistedServiceConfig struct {
+	ServiceID string
+	Defaults  *persistedServiceConfigResponse
+}
+
+// persistedServiceConfigResponse is a mirror image of
+// *structs.ServiceConfigResponse without QueryMeta fields. These should be
+// kept in sync.
```

Comment:
Why can't we use the existing struct? Does it add empty QueryMeta fields to the JSON we persist? I'm OK with this just not super sure why it's important.

Reply:
We could use the existing struct, it just seemed a little weird to persist all of that metadata with the original values. I can revert this section and fixup the tests.